Multi-shipping with two consignments

andresa-razoyo opened the pull request.

andresa-razoyo committed:
Multi-shipping with two consignments

andresa-razoyo committed:
Add back missing code

andresa-razoyo committed:
Hide all sections when FFL dealer is not selected

edias-razoyo commented on public/js/bigcommerce.js:
```diff
@@ -17,7 +18,7 @@ let FFLConfigs = {
     platform: 'BigCommerce',
     automaticFFLStoreInfoEndpointUrl: 'https://app-stage.automaticffl.com/store-front/api/stores/',
     // automaticFFLIframeUrl: 'https://automaticffl.pages.dev',
-    automaticFFLIframeUrl: 'http://localhost:3000',
```
I think that this URL should be:
https://static-stage.automaticffl.com/big-commerce-enhanced-checkout

andresa-razoyo replied:
This is a parameter that should have been setup with William. Didn't you do that a few weeks ago?

edias-razoyo commented on public/js/bigcommerce.js:
```diff
@@ -425,13 +461,152 @@ async function setShippingConsignments(dealerData) {
+                            lastName: "",
+                            company: '',
+                            address1: '',
+                            city: ' ',
```
There is a space here. I'm not sure, but maybe when the client starts typing the city, the space remains and might be annoying.

andresa-razoyo replied:
This doesn't go to the input field, it is just for the consignment. The empty space is necessary, otherwise the API will just ignore this consignment and keep everything in a single one.

When the customer goes to the checkout, thanks to this, a message saying the address is invalid is displayed, the user needs to open the select box and use the option to add a new address, which will open an empty address form.

edias-razoyo commented on public/js/bigcommerce.js:
```diff
@@ -453,18 +628,25 @@ async function initFFLProducts() {
+            filteredProducts.others.push(fullProductData);
             return;
         }
         FFLConfigs.hasNonFFLProducts = true;
```
Giancarlo informed me that if the "FFL" attribute has the value "YES", we should also consider it as a firearm. Additionally, the line "FFLConfigs.hasNonFFLProducts = true;" is no longer accessible, which might cause some side effects.

I believe the ideal approach would be:
`        if (fflTypeField && fflTypeField.node.value.toLowerCase() === 'ammo') {
            filteredProducts.ammo.push(fullProductData);
            return;
        } else if ((fflTypeField && fflTypeField.node.value.toLowerCase() === 'firearm') ||
            (fflField && fflField.node.value.toLowerCase() === 'yes')) {
            filteredProducts.fireArm.push(fullProductData);
            return;
        } else {
            FFLConfigs.hasNonFFLProducts = true;
            filteredProducts.others.push(fullProductData);
            return;
        }`

edias-razoyo commented on src/components/Dealer/Locator.tsx:
```diff
@@ -59,7 +59,7 @@ export default class Locator extends React.PureComponent<LocatorProps, LocatorSt
     if (location && radius) {
       this.setState({loading: true, currentLocation: location});
-      fetch(`https://${process.env.REACT_APP_HOST}/store-front/api/${this.props.storeHash}/dealers?location=${location}&radius=${radius}`,
+      fetch(`https://app-stage.automaticffl.com/store-front/api/${this.props.storeHash}/dealers?location=${location}&radius=${radius}`,
```
Does the production and stage use the same URL, or must it be dynamic?

andresa-razoyo replied:
Forgot about it, removed.

edias-razoyo commented on src/components/Dealer/Locator.tsx:
```diff
@@ -171,7 +171,7 @@ export default class Locator extends React.PureComponent<LocatorProps, LocatorSt
             <div className='hidden lg:block w-3 h-full absolute bg-gradient-to-r from-gray-300 to-transparent z-20'></div>
             <LocatorMap
-              apiKey={ this.props.googleMapsApiKey }
+              apiKey="AIzaSyCOk4pPoys3i373bRTQq9VrMC2UGWV574k"
```
Should it be hardcoded?

andresa-razoyo replied:
Forgot this one too

andresa-razoyo committed:
Add back removed code

andresa-razoyo committed:
Update "hasNonFFLProducts" condition

FFL

andresa-razoyo committed:
Generate config.js file with environment variables

Generate config.js file with environment variables

Generate config.js file with environment variables

andresa-razoyo committed:
Update README.md

andresa-razoyo committed:
Added environment substitutions
